packages/grid/x-data-grid-pro/src/hooks/features/columnResize/useGridColumnResize.tsx

```diff
@@ -426,6 +432,41 @@ export const useGridColumnResize = (
     apiRef.current.forceUpdate();
   }, [apiRef]);
 
+  /**
+   * API METHODS
+   */
+
+  const autosizeColumns = React.useCallback<GridColumnResizeApi['autosizeColumns']>(
+    async (options = {}) => {
+      const root = apiRef.current.rootElementRef?.current;
+      if (!root) {
+        return;
+      }
+
+      const context = await disableColumnVirtualization(apiRef);
+
+      const [columns, widthsByField] = extractColumnWidths(apiRef, root, options);
+
+      await restoreColumnVirtualization(apiRef, context);
+
+      const newColumns = columns.map((column) => {
+        const newColumn = { ...column };
+        const width = widthsByField[column.field];
+        newColumn.width = clamp(width.contentMax, width.min, width.max);
+        return newColumn;
+      });
```

Comment:
Right now there is only the naive implementation, every column gets the width of its max content width. Do we want to be cleverer here?

One option that I see could be interesting would be an option to try to fit the columns as much as possible inside the grid viewport.

Another interesting options could be to remove outliers, such that this outlying value wouldn't cause the full column to waste a lot of space:

```
| Header                 |
| ---------------------- |
| a                      |
| b                      |
| c                      |
| d                      |
| outlyingly large value |
| e                      |
| f                      |
```

Reply:
I've added the option, play with it [in the demo](https://deploy-preview-10180--material-ui-x.netlify.app/x/react-data-grid/column-dimensions/#autosizing).

Reply:
> One option that I see could be interesting would be an option to try to fit the columns as much as possible inside the grid viewport.

This one sounds like a different feature that contradicts column autosize because you might need to shrink columns to fit them into the grid viewport.
What do you think?

Reply:
Yes I was thinking of the case when the columns width is smaller than the available width, it indeed doesn't make sense if it's bigger.

Reply:
So the columns will be wider than the content to fully use the grid viewport?

Reply:
Yes. I'll add an option to play with it, it's a trivial calculation.

Reply:
Added.